### PR TITLE
scalariform props file absolute/relative path flag

### PR DESCRIPTION
**Pull request by pbassiner**
Scalariform properties files are usually project dependent and I think atom-scalariform could support it by having a absolute/relative flag regarding the properties file path.

I've created a new property `relativeToProject` which, if set to true, assumes the `propertiesFile` path is relative to the about-to-format file's project folder.

### CODE REVIEW

File: lib/atom-scalariform.js [hunk 1]
```diff
@@ -5,6 +5,19 @@ var packagePath = function() {
   return atom.packages.getLoadedPackage('atom-scalariform').path;
 }
 
+var projectPath = function(fileToFormatPath) {
+  var ref = atom.project.getDirectories();
+  var projectPath = '';
+  for (var i = 0, len = ref.length; i < len; i++) {
+    var directory = ref[i];
+    if (fileToFormatPath.indexOf(directory.path) > -1) {
+            projectPath = directory.path;
```
Comment (jackhopner) [inline lib/atom-scalariform.js]: break out of for loop here?

**Comment by jackhopner** [comment]
So the aim of this would be to be able to support a properties file in each package? Does it need a fallback to the root project folder if it can't find one?

File: lib/atom-scalariform.js [hunk 2]
```diff
@@ -5,6 +5,19 @@ var packagePath = function() {
   return atom.packages.getLoadedPackage('atom-scalariform').path;
 }
 
+var projectPath = function(fileToFormatPath) {
+  var ref = atom.project.getDirectories();
+  var projectPath = '';
+  for (var i = 0, len = ref.length; i < len; i++) {
+    var directory = ref[i];
+    if (fileToFormatPath.indexOf(directory.path) > -1) {
```
Comment (jackhopner) [inline lib/atom-scalariform.js]: Probably need to do something like 

`fileToFormatPath.substring(0, fileToFormatPath.lastIndexOf('/')) === directory.path`

Otherwise abc/def/Main.scala would match if you also had a directory abc/ghi/Test.scala?

**Review by jackhopner — COMMENTED** [review]
Couple of comments, also will need to resolve the merge conflicts. Thanks for your contributions so far!

File: lib/atom-scalariform.js (hunk 2 above)
Comment (pbassiner) [reply lib/atom-scalariform.js]: Not really, `directory.path` will only include the absolute path to the project folder, but the `scala` file to format might be in a subdirectory there:
* Project: /home/user/myProject
* File: /home/user/myProject/folder1/subfolder1/Test.scala

The only way that you could get a false positive would be:
* Actual Project: /home/user/myProject
* Other Project: /home/user/myOtherProject
* File: /home/user/myProject/home/user/myOtherProject/Test.scala

Am I missing any other corner case?

Maybe I just need to force `indexOf` to be `0`:
```js
fileToFormatPath.indexOf(directory.path) === 0
```

What do you think?

File: lib/atom-scalariform.js (hunk 1 above)
Comment (pbassiner) [reply lib/atom-scalariform.js]: I can `return` directly, right.

**Comment by pbassiner** [comment]
The aim of the PR is indeed supporting a properties file in each project.

Regarding the fallback, the only way I've been able to make the plugin not find any matching project is removing all projects from the sidebar and keeping the file to format open. Then there's no project folder at all, so I think the fallback could be to consider that the path to the properties file is an absolute path anyway. Do you have a better approach in mind?

File: lib/atom-scalariform.js (hunk 2 above)
Comment (jackhopner) [reply lib/atom-scalariform.js]: What about:

- Project: /home/user/myProject
- File: /home/user/myProject/abc/def/Main.scala
- File: /home/user/myProject/abc/Test.scala

When formatting Main.scala the `fileToFormatPath` would be /home/user/myProject/abc/def/Main.scala correct? That means if you it checked if the index of  /home/user/myProject/abc/ was > 1 it would return true as that index would actually be 0?

File: lib/atom-scalariform.js (hunk 1 above)
Comment (jackhopner) [reply lib/atom-scalariform.js]: yeah sorry thats better

File: lib/atom-scalariform.js (hunk 2 above)
Comment (jackhopner) [reply lib/atom-scalariform.js]: Just reading your comments again on the PR I think I may have misunderstood the purpose of this PR, the check for === 0 should be fine.

**Comment by jackhopner** [comment]
That sounds fine @pbassiner

**Comment by pbassiner** [comment]
Hi @jackhopner,
I've applied the discussed modifications in a new commit (will fixup when you consider it's ready to be merged).

I just thought about an alternative implementation though: what about removing the `relativeToProject` flag and simply consider the `propertiesFile` an absolute path if starts with a `/` and a relative path otherwise? It simplifies the configuration management, although it's certainly not so obvious for the user. What do you think?

**Comment by jackhopner** [comment]
sounds good @pbassiner

**Comment by pbassiner** [comment]
Hi @jackhopner,
as discussed, I've removed the `relativeToProject` flag and updated the `README` accordingly.

Thanks!

**Comment by jackhopner** [comment]
Thanks @pbassiner, I'll update the scalariform jar and make a release.
